### PR TITLE
Run terminate handler when command is finished or cancelled

### LINKED ISSUE #416: Handle CTRL-C gracefully in WP CLI
## Description

We need to handle when a user cancels a long-running WP CLI command

- Forward the Ctrl-C character (x03) to the command runner 
- Call an endpoint in the API to mark the command as cancelled (and do any other cleanup)

### PR DESCRIPTION
**Pull request by dchymko**
## Description
When a command is prematurely terminated (via CTRL-C), we need to run cleanup tasks such as notifying the command runner to stop the commands.

## Steps to Test
1. Pull down PR and related API PR
2. Run `wp user create --prompt` and wait for the first prompt
3. Ctrl-C to exit
4. Verify it was cancelled (in the app dashboard)
5. Repeat for `wp shell`
6. Repeat for the psuedo shell
7. Try a few "long-running" commands (may be difficult to hit the timing on small sites) `wp option list` `wp jetpack sync start`

Note that for commands with a lot of output (`wp option list` for example), the command may still stream output text for a second or two after it completes. So if you cancel it at that point, it will show as 'completed' since it did actually finish running.

Closes #416

### CODE REVIEW

File: src/bin/vip-wp.js
Comment (darthhexx) [inline src/bin/vip-wp.js]: Please just send the '\x03' character, without the newline, and you will be in business :) The runner is now expecting a data packet of length 1 with just the value of 0x3.

File: src/bin/vip-wp.js
Comment (elazzabi) [inline src/bin/vip-wp.js]: can we make the signature more explicit here? The difference between `shutdownHandler( true, guid )` and `shutdownHandler( false, guid )` isn't explicit until you visit the function definition.

we can opt for: `shutdownHandler( guid, { exitProcess: true } )` or something similar

File: src/bin/vip-wp.js
Comment (elazzabi) [inline src/bin/vip-wp.js]: I guess leaving this `process.exit();` is better than wrapping it with `shutdownHandler( false );`. Especially if we aren't adding anything in `shutdownHandler` other than calling `process.exit();` inside (which is the current case when passing `false`)

File: src/bin/vip-wp.js
Comment (elazzabi) [inline src/bin/vip-wp.js:178]: I think we can use `cliCommand.guid` directly 🤔

**Review by elazzabi — CHANGES_REQUESTED** [review]
Looks good, with some minor changes.

+ We should make sure we remove the `console.log` before merging

**Comment by dchymko** [comment]
Updated the test steps with some examples.

File: src/bin/vip-wp.js
Comment (dchymko) [reply src/bin/vip-wp.js:178]: cliCommand is defined inside the `subShellRl.on( 'line', async line => {` event handler, but isn't available in the `subShellRl.on( 'SIGINT', () => {` handler,  hence setting a variable outside of that.

File: src/bin/vip-wp.js
Comment (dchymko) [reply src/bin/vip-wp.js]: Given your other comment about leaving the process.exit() I just removed that additional parameter from the method.  I had envisioned using the shutdownHandler in other cases and was trying to make it more generic.  If we need to use it in multiple places, we can revisit, but for now simplification is better.

**Review by elazzabi — APPROVED** [review]

**Comment by elazzabi** [comment]
Thank you for working on this @dchymko ! Looks good to me (minus the `console.log`s), I'll run some tests and report back

**Comment by elazzabi** [comment]
Tested locally with all cases in the test steps, looks good!

**Comment by dchymko** [comment]
Thanks for testing @elazzabi. The console.logs that are currently there are intentional and match other flows we have. I assume you were referring to the one removed in 2113c81?

I'll hold off merging this until the endpoint is in Automattic/vip-go-api-public#653

**Comment by dchymko** [comment]
Thanks for testing @elazzabi. The console.logs that are currently there are intentional and match other flows we have. I assume you were referring to the one removed in 2113c81?

I'll hold off merging this until the endpoint has been merged in.

**Comment by elazzabi** [comment]
@dchymko oh yes, I was referring to those. All looks good 👍

**Pull request merged**
